src/directives/formly-form.controller.js

```diff
@@ -30,7 +30,7 @@ export default function FormlyFormController(
   function validateFormControl(formControl, promise) {
     const validate = formControl.$validate
     if (promise) {
-      promise.then(validate)
+      promise.then(() => validate.apply(formControl))
     } else {
       validate()
```

Comment:
I was thinking that the problem is that the `validate` function isn't being called with the right `this` context. Honestly, I'd prefer the use of `bind` personally, but I don't care either way.